### PR TITLE
docs(contributor-guide): update v0.7 indexing

### PR DESCRIPTION
**Pull request by zhongzc**
## What's Changed in this PR

as title

## Checklist

- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 1]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+Search engines and time series databases belong to two different domains. Although the principle of the inverted index technology applied is similar, there are some conceptual migrations:
+1. Term: In GreptimeDB, it refers to the column value of the time series.
+2. Document: In GreptimeDB, it refers to the data segment containing multiple time series.
+
+The introduction of the inverted index allows GreptimeDB to skip data segments that do not meet the conditions during queries, thereby improving scanning performance.
+
+![Inverted index searching](/inverted-index-searching.png)
+
+For example, in the above query, retrieve the data segments where `job` = `apiserver`, `handler` matches the regex `.*users`, and `status` matches the regex `4..` from the inverted index, and get the data segment list `[1]`. Subsequently, only the data segment needs to be scanned, reducing a large number of IO operations.
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:55]:
```suggestion
For instance, the query above uses the inverted index to identify data segments where `job` equals `apiserver`, `handler` matches the regex `.*users`, and `status` matches the regex `4...`. It then scans these data segments to produce the final results that meet all conditions, significantly reducing the number of IO operations.
```

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 2]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+### Inverted Index Format
+
+The inverted index is stored by column internally, and each column's inverted index block consists of FST and a Bitmap storing data segment IDs.
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:59]: This description confuses user, but I don't know how to modify it...

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 3]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+The FST (Finite State Transducer) allows us to store mappings from column values to Bitmap positions in a compact format and provides excellent search performance and complex search capabilities (such as regular expression matching).
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:61]:
```suggestion
The FST (Finite State Transducer) allows GreptimeDB to store mappings from column values to Bitmap positions in a compact format and provides excellent search performance and supports complex search capabilities (such as regular expression matching).
```

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 4]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+The Bitmap is used to store data segment IDs, with each bit representing a data segment, and a 1 indicates that the data segment contains the column value.
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:63]: It's also confusing here, where does `1` come from? what is `the column value`?

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 5]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+![Inverted index format](/inverted-index-format.png)
+
+### Index Data Segments
+
+GreptimeDB divides an SST file into several indexed data segments, each containing an equal number of rows.
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:69]:
```suggestion
GreptimeDB divides an SST file into multiple indexed data segments, with each segment housing an equal number of rows.
This segmentation is designed to optimize query performance by scanning only the data segments that match the query conditions. 
```

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md [hunk 6]
```diff
@@ -28,11 +28,54 @@ Apache Parquet file format provides inherent statistics in headers of column chu
...
+
+For example, if a data segment contains 1024 rows and the data segment list obtained after applying the inverted index to the query conditions is `[0, 2]`, then only the 0th and 2nd data segments in the SST file need to be scanned, i.e., from row 0 to row 1023 and from row 2048 to row 3071.
```
Comment (nicecui) [inline docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md:71]:
```suggestion
For example, if a data segment contains 1024 rows and the list of data segments identified through the inverted index for the query conditions is `[0, 2]`, then only the 0th and 2nd data segments in the SST file—from rows 0 to 1023 and 2048 to 3071, respectively—need to be scanned.
```

File: docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md (hunk 2 above)
Comment (zhongzc) [reply docs/v0.7/en/contributor-guide/datanode/data-persistence-indexing.md]: I rephrased it, please take a look.

**Review by nicecui — APPROVED** [review]

**Pull request merged**